### ISSUE TITLE
Wrong auto correction Style/SoleNestedConditional

### ISSUE THREAD

**Issue by thinca** [rubocop-hq/rubocop#9233]
I got a wrong auto-correction with `Style/SoleNestedConditional` cop and without `Style/IfUnlessModifier` cop.

`.rubocop.yml`
```yaml
AllCops:
  NewCops: disable

Style/IfUnlessModifier:
  Enabled: false
```

`test.rb`
```ruby
# an example file

def func
  # foo
  foo

  # hoge
  if foo
    if bar
      buz
    end
  end
end
```

Run:
```
$ rubocop -a --debug --only Style/SoleNestedConditional test.rb
```

## Expected behavior

```ruby
# an example file

def func
  # foo
  foo

  # hoge
  if foo && bar
    buz
  end
end
```

## Actual behavior

Output:
```
For /Users/thinca/tmp/rubocop-test: configuration from /Users/thinca/tmp/rubocop-test/.rubocop.yml
Default configuration from /Users/thinca/tmp/rubocop-test/vendor/bundle/ruby/2.7.0/gems/rubocop-1.6.1/config/default.yml
Inspecting 1 file
Scanning /Users/thinca/tmp/rubocop-test/test.rb
Loading cache from /Users/thinca/.cache/rubocop_cache/fd9e92937dd32541de61425da45d7dfa0c54bb9e/5c3a0012aae88f33b221aa9bbc6dcfba8f2100c7/9f77710eeb664a239190bf981ba7f5ff6c702a02
C

Offenses:

test.rb:9:5: C: [Corrected] Style/SoleNestedConditional: Consider merging nested conditions into outer if conditions.
    if bar
    ^^

1 file inspected, 1 offense detected, 1 offense corrected
Finished in 0.12683500000275671 seconds
```

`test.rb`
```ruby
# an example file

def func
  # foo
  foo

  # hoge
  # an example file
# foo
# hoge
if foo && bar
      buz
    end
end
```

- Extra comments are appears
- Strange indentation

## Steps to reproduce the problem

```
$ mkdir -p ~/tmp/rubocop-test
$ cd ~/tmp/rubocop-test
$ (put .rubocop.yml and test.rb. rubocop is already installed.)
$ rubocop -a --debug --only Style/SoleNestedConditional test.rb
```

## RuboCop version

```
$ rubocop -V
1.6.1 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 2.7.2 x86_64-darwin19)
```

**Comment by koic** [comment]
Thank you for the feedback. Part of auto-correction of indentation will be left to other `Layout` cop yet (`Layout/IndentationWidth` cop), but the auto-corrected comment is clearly incorrect.
